src/webportal/src/app/job/job-view/fabric/job-detail/components/task-role-container-list.jsx

```diff
@@ -806,72 +674,222 @@ export default class TaskRoleContainerList extends React.Component {
           </div>
         ),
       },
+      {
+        key: 'gpus',
+        name: 'GPUs',
+        className: FontClassNames.mediumPlus,
+        headerClassName: FontClassNames.medium,
+        minWidth: 35,
+        maxWidth: 60,
+        isResizable: true,
+        onRender: item => {
+          const gpuAttr = isNil(item.containerGpus)
+            ? null
+            : parseGpuAttr(item.containerGpus);
+          if (isNil(gpuAttr)) {
+            return null;
+          } else if (gpuAttr.length === 0) {
+            return <div>0</div>;
+          } else {
+            return (
+              <div>
+                <TooltipHost
+                  calloutProps={{
+                    isBeakVisible: false,
+                  }}
+                  tooltipProps={{
+                    onRenderContent: () => (
+                      <div>
+                        {gpuAttr.map(x => (
+                          <span
+                            className={t.mr2}
+                            key={`gpu-${x}`}
+                          >{`#${x}`}</span>
+                        ))}
+                      </div>
+                    ),
+                  }}
+                  directionalHint={DirectionalHint.topLeftEdge}
+                >
+                  <Stack horizontal gap='s1'>
+                    <div>{gpuAttr.length}</div>
+                    <div>
+                      <Icon
+                        iconName='Info'
+                        styles={{
+                          root: [
+                            { fontSize: FontSizes.small },
+                            ColorClassNames.neutralSecondary,
+                          ],
+                        }}
+                      />
+                    </div>
+                  </Stack>
+                </TooltipHost>
+              </div>
+            );
+          }
+        },
+      },
+      {
+        key: 'exitCode',
+        name: 'Exit Code',
+        headerClassName: FontClassNames.medium,
+        minWidth: 100,
+        maxWidth: 150,
+        isResizable: true,
+        onRender: item => {
+          return (
+            <div className={c(FontClassNames.mediumPlus)}>
+              {item.containerExitCode}
+            </div>
+          );
+        },
+      },
+    ];
+    const optionalColumns = [
+      {
+        key: 'startTime',
+        name: 'Start Time',
+        headerClassName: FontClassNames.medium,
+        minWidth: 150,
+        maxWidth: 200,
+        isResizable: true,
+        onRender: item => {
+          return (
+            <div className={c(FontClassNames.mediumPlus)}>
+              {isNil(item.createdTime)
+                ? 'N/A'
+                : printDateTime(DateTime.fromMillis(item.createdTime))}
+            </div>
+          );
+        },
+      },
+      {
+        key: 'taskAttemptCompletedTime',
+        name: 'Task Attempt Completed Time',
+        headerClassName: FontClassNames.medium,
```

Comment:
Change to duration to align with other parts?

Reply:
Same reason, I think in the column we should present the raw info to user from api instead of processing it to duration.

Reply:
But how user sort the col to find the max run duration task?

---
In reply to: [506073187](https://github.com/microsoft/pai/pull/4966#discussion_r506073187) [](ancestors = 506073187)

Reply:
I see job list page also use duration, should we align with it?

---
In reply to: [507486610](https://github.com/microsoft/pai/pull/4966#discussion_r507486610) [](ancestors = 507486610,506073187)